library/core/src/char/mod.rs

```diff
@@ -97,6 +97,17 @@ pub const REPLACEMENT_CHARACTER: char = char::REPLACEMENT_CHARACTER;
 #[stable(feature = "unicode_version", since = "1.45.0")]
 pub const UNICODE_VERSION: (u8, u8, u8) = char::UNICODE_VERSION;
 
+/// The maximum number of bytes required to [encode](char::encode_utf8) a
+/// `char` in UTF-8. Use [`char::MAX_UTF8_LEN`] instead.
+#[stable(feature = "max_len", since = "1.63.0")]
+pub const MAX_UTF8_LEN: usize = char::MAX_UTF8_LEN;
```

Comment:
pondering: are the module-level constants still desirable, now that we can have type-associated ones?  It's unclear to me how to apply precedent from #68490 here -- the `char` module certainly can't go away.  But if the doc comment on this constant says not to use it, why add it?

Reply:
yeh i didn't think of type-associated ones, will change it to that